apps/omg_api/lib/state/core.ex

```diff
@@ -151,8 +151,9 @@ defmodule OMG.API.State.Core do
 
     with :ok <- validate_block_size(state),
          {:ok, input_amounts_by_currency} <- correct_inputs?(state, recovered_tx),
-         output_amounts_by_currency <- get_amounts_by_currency(outputs),
-         :ok <- amounts_add_up?(input_amounts_by_currency, output_amounts_by_currency, fees) do
+         output_amounts_by_currency = get_amounts_by_currency(outputs),
+         transaction_fees = Transaction.Fee.apply(recovered_tx, Map.keys(input_amounts_by_currency), fees),
```

Comment:
oh I see now, this is where that `Map.take` ended up in. This is not good, as this is calculated single-threaded, so we're bleeding performance out of our critical tx-handling path. I say the `Map.take` goes back to its original spot